frontend/src/app/app.component.ts

```diff
@@ -46,8 +46,9 @@ export class AppComponent implements OnInit {
           localStorage.removeItem('showLoggedIn');
           this.setStatus();
         }
-        ga('set', 'page', event.urlAfterRedirects);
-        ga('send', 'pageview');
+
+        (<any>window).ga('set', 'page', event.urlAfterRedirects);
```

Comment:
fwiw, that's what I had before and it was still failing. I'm just not sure whether its a build problem or a test setting problem.

Reply:
hmmm, even declaring var ga as a function in app.component it's failing the same tests as undefined. Will keep digging.